packages/contracts-bedrock/test/governance/ProposalValidator.t.sol

```diff
@@ -578,11 +656,497 @@ contract ProposalValidator_HashProposalWithModule_Test is ProposalValidator_Init
+            abi.encodeCall(IOptimismGovernor.proposalSnapshot, (uint256(expectedHash))),
+            abi.encode(0)
+        );
+
+        vm.prank(rando);
+        bytes32 proposalHash = validator.submitFundingProposal(
+            criteriaValue,
+            optionsDescriptions,
+            optionsRecipients,
+            optionsAmounts,
+            description,
+            ProposalValidator.ProposalType.GovernanceFund
+        );
+
+        assertTrue(proposalHash != bytes32(0));
+    }
+
+    function test_submitFundingProposal_zeroAmount_succeeds() public {
+        optionsAmounts[0] = 0;
+        optionsAmounts[1] = 100 ether;
+
+        // Calculate expected proposal hash
+        bytes memory votingModuleData =
+            _constructVotingModuleData(optionsDescriptions, optionsRecipients, optionsAmounts, criteriaValue);
+        bytes32 expectedHash = validator.hashProposalWithModule(
+            makeAddr("approvalVotingModule"), votingModuleData, keccak256(bytes(description))
+        );
+
+        // Mock proposalSnapshot to return 0 for the expected proposal hash
+        _mockAndExpect(
+            address(governor),
+            abi.encodeCall(IOptimismGovernor.proposalSnapshot, (uint256(expectedHash))),
+            abi.encode(0)
+        );
+
+        vm.prank(rando);
+        bytes32 proposalHash = validator.submitFundingProposal(
+            criteriaValue,
+            optionsDescriptions,
+            optionsRecipients,
+            optionsAmounts,
+            description,
+            ProposalValidator.ProposalType.GovernanceFund
+        );
+
+        assertTrue(proposalHash != bytes32(0));
+    }
+
+
+}
+
+/// @title ProposalValidator_SubmitFundingProposal_TestFail
+/// @notice Sad path tests for submitFundingProposal function
+contract ProposalValidator_SubmitFundingProposal_TestFail is ProposalValidator_Init {
+    uint128 criteriaValue;
+    string[] optionsDescriptions;
+    address[] optionsRecipients;
+    uint256[] optionsAmounts;
+    string description;
+
+    function setUp() public override {
+        super.setUp();
+
+        // Set GovernanceFund to use the approval voting module
+        vm.prank(owner);
+        validator.setProposalTypeData(
+            ProposalValidator.ProposalType.GovernanceFund,
+            ProposalValidator.ProposalTypeData({
+                requiredApprovals: PROPOSAL_REQUIRED_APPROVALS,
+                proposalVotingModule: 3
+            })
+        );
+
+        criteriaValue = 50;
+        optionsDescriptions = new string[](2);
+        optionsDescriptions[0] = "Option A";
+        optionsDescriptions[1] = "Option B";
+
+        optionsRecipients = new address[](2);
+        optionsRecipients[0] = makeAddr("recipient1");
+        optionsRecipients[1] = makeAddr("recipient2");
+
+        optionsAmounts = new uint256[](2);
+        optionsAmounts[0] = 1000 ether;
+        optionsAmounts[1] = 500 ether;
+
+        description = "Test funding proposal";
```

Comment:
This setup is repeated in `ProposalValidator_SubmitFundingProposal_Test` aswell, maybe defining the proposal type data is something that could be added in the `Init` contract since this will probably be needed to happen in most of the tests?

Reply:
I think with the usage of `_setGovernanceFundProposalType` its enough, the rest of the hardcoded values should be fuzzed where possible